### PR TITLE
Mkarnowski vmc

### PR DESCRIPTION
**Pull request by mkarnowski**
playbooks for Controller VM deployment/3node cluster creation and SE deployment for VMC added to ansible directory.

### CODE REVIEW

File: ansible/vmc/vmc-ctl.yml [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+- hosts: localhost
+  connection: local
+  roles:
+    - avinetworks.avisdk
+  vars_files:
+    - "./vmc-ctl-vars.yml"
+
+  tasks:
+
+    - name: Create temp SSH Key directory
```
Comment (grastogi23) [inline ansible/vmc/vmc-ctl.yml:11]: I would suggest putting a more structured names like

Avi Install: Ssh Setup

File: ansible/vmc/vmc-ctl.yml [hunk 2]
```diff
@@ -0,0 +1,575 @@
...
+      file:
+       path: ./avikey
+       state: directory
+  
+    -  stat:
+         path: ./avikey/avi_id_ssh_rsa
+       register: stat_result
+
+
+    - name: Generate SSH Key
```
Comment (grastogi23) [inline ansible/vmc/vmc-ctl.yml:21]: Add Avi Install: Ssh Setup: Generate new SSH Key

File: ansible/vmc/vmc-ctl.yml [hunk 3]
```diff
@@ -0,0 +1,575 @@
...
+
+    - set_fact:
+        pub_key: "{{ lookup('file', './avikey/avi_id_ssh_rsa.pub') }}"
+
+
+    #- debug:
+    #    msg: "{{ pub_key }}"
+
+
+    - name:  deploy Avi Controller 1
```
Comment (grastogi23) [inline ansible/vmc/vmc-ctl.yml:36]: change the name. Avi Controller: Deploy Single Node Cluster. can we add check for controller if it already exists?

File: ansible/vmc/vmc-ctl.yml [hunk 4]
```diff
@@ -0,0 +1,575 @@
...
+        ctl1vm_uuid: "{{ ctl1_status.instance.hw_product_uuid }}"
+      when:  "ctl1_status is defined and 'instance' in ctl1_status"
+
+
+    - set_fact:
+        ctl1vm_uuid: "{{ ctl1_status.hw_product_uuid }}"
+      when:  "ctl1_status is defined and 'hw_product_uuid' in ctl1_status"
+
+
+    - name: Modify Controller 1 Hardware
```
Comment (grastogi23) [inline ansible/vmc/vmc-ctl.yml:70]: change the heading.

File: ansible/vmc/vmc-ctl.yml [hunk 5]
```diff
@@ -0,0 +1,575 @@
...
+        hardware:
+          memory_mb: "{{ se_memory_mb | default(24576,true) }}"
+          num_cpus: "{{ se_num_cpus | default(8,true) }}"
+        disk:
+        - size_gb: "{{ se_disk_size_gb | default(128,true) }}"
+        state: present
+      delegate_to: "localhost" 
+
+
+    - name:  deploy Avi Controller 2
```
Comment (grastogi23) [inline ansible/vmc/vmc-ctl.yml:86]: can this be all parameterized to be in a loop. We are doing the same thing for every controller. There used to be an example of this.

**Review by grastogi23 — COMMENTED** [review]
We already had some ansible playbooks to bring up cluster. let us get the code from them to use the loop rather than individual controllers.

there was also cde to deal with controller not bein up.

**Comment by mkarnowski** [comment]
closing pr

**Pull request closed without merging**